Lesson15/Busra/busra.js

```diff
@@ -0,0 +1,508 @@
+*/
+function checkFileExtension(filename){
+  if(filename.endsWith(".pdf")){
+    console.log("This is a PDF file.");
+  }
+  else{
+    console.log("Not a PDF file.");
+  }
+}
+checkFileExtension("dosya.pdf");
+checkFileExtension("dosya.jpeg");
+
+
+/*
+8. Palindrome Check
+   - Define a function `isPalindrome(str)` that checks if `str` is the same
+     forwards and backwards (normalize the string and compare it to reversed version of itself).
+   - If it is, log: "<str> is a palindrome"
+   - Otherwise, log: "<str> is not a palindrome"
+*/
+function isPalindrome(str) {
+  
+  let normalizeStr = str.toLowerCase().replace(/[^a-z0-9]/g, "");// büyük harf ve karakterli stringi normalize eder
+
+
+  let ters = normalizeStr.split('').reverse().join('');
+
+
+  if (normalizeStr === ters) {
+    console.log(str +  "  is a palindrome");
+  } else {
+    console.log(str + "  is not a palindrome");
+  }
+}
+
+isPalindrome("busra");
+isPalindrome("ey Nihat Adana'da tahin ye");  //Adana ve Nihat isimlerini büyük harfle yazdığımızda, Adana'da da kesme işareti kullandığımızda doğru sonucu vermiyor bu yüzden en başta bir düzenleme yapmamız gerek
+
+
+
+
+
+/*
+10. Check Even or Odd (if-else)
+   - Define a function `evenOrOdd(number)` that:
+     - Logs "Even" if the number is even
+     - Logs "Odd" if the number is odd
+*/
+
+function evenOrOdd(number){
+
+  if(number%2===0){
+    console.log("number is even");
+  }
+  else{
+    console.log("number is odd");
+  }
+}
+
+evenOrOdd(224);
+evenOrOdd(7485);
+
+
+
+/*
+11. URL Protocol Checker
+   - Define a function `checkProtocol(url)` that converts the URL to lowercase
+     and checks if it starts with "https" using .startsWith().
+   - Log "Secure connection" if true, otherwise "Unsecure connection".
+*/
+
+function checkProtocol(url){
+  let normalizeUrl = url.toLowerCase();
+  if(normalizeUrl.startsWith("https")){
+    console.log("Secure connection");
+  }
+  else{
+    console.log("Unsecure connection");
+  }
+
+}
+checkProtocol("https://www.freecodecamp.org/");
+
+checkProtocol("hptts://www.freecodecamp.org/");
+
+
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+
+function repeatWord(word, times){
+ 
+  console.log(word.repeat(times));
+}
+repeatWord("busra",15);
+
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+function censorWord(sentence, target){
+  console.log(sentence.replace(target,"****"));
+}
+censorWord("I need to study","need");
+
+
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+
+function startsWithA(str){
+  if(str.charAt(0)==='A'){
```

Comment:
It could be that the first letter is in lower case so always safer to convert the incoming value to the same format as the value that you compare it to (unless you are doing a case-sensitive comparison).
```suggestion
  if(str.charAt(0).toUpperCase() ==='A'){
```